### PR TITLE
Graduate NonPreemptingPriority to GA

### PR DESCRIPTION
**Pull request by denkensk**
#### What type of PR is this?
/kind feature
/sig scheduling

#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/enhancements/issues/902

#### Does this PR introduce a user-facing change?

```release-note
Feature of `NonPreemptingPriority` is graduated  to GA
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/902-non-preempting-priorityclass
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
/hold
for kubernetes/enhancements#3110

File: pkg/registry/scheduling/priorityclass/strategy.go
Comment (kerthcet) [inline pkg/registry/scheduling/priorityclass/strategy.go:48]: I was wondering why we'd like to remove these codes, it doesn't look like a problem.

Comment (denkensk) [reply pkg/registry/scheduling/priorityclass/strategy.go:48]: Thanks. Restore the initialization of generation value

**Comment by denkensk** [comment]
/unhold
Because the code was merged.

**Comment by denkensk** [comment]
> Did you already include the change to the event message? Although it might be better to have in a separate PR that get's merged before this one.

It isn’t included in this pr. I will sent another pr for the event message updating. This GA one can be merged after that one.

File: pkg/registry/scheduling/priorityclass/strategy.go
Comment (Huang-Wei) [inline pkg/registry/scheduling/priorityclass/strategy.go:57]: Just FYI for @ravisantoshgudimetla: once you start to work on mutable PriorityClass, need to drop in logic to update the generation by 1 here.

File: plugin/pkg/admission/priority/admission.go
Comment (Huang-Wei) [inline plugin/pkg/admission/priority/admission.go:82]: I think you can remove this, along with L61.

**Review by Huang-Wei — COMMENTED** [review]
Some nits. Thanks @denkensk .

**Comment by Huang-Wei** [comment]
> Did you already include the change to the event message?

@alculquicondor what's this message about? Indicate a unschedulable pod is PreemptNever Or PreemptLowerPriority for troubleshooting?

**Comment by alculquicondor** [comment]
https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/902-non-preempting-priorityclass#stable-v124

**Comment by denkensk** [comment]
> https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/902-non-preempting-priorityclass#stable-v124

@Huang-Wei @alculquicondor  
add pr for this
https://github.com/kubernetes/kubernetes/pull/107775

File: plugin/pkg/admission/priority/admission.go
Comment (denkensk) [reply plugin/pkg/admission/priority/admission.go]: thanks  Done

**Review by ravisantoshgudimetla — COMMENTED** [review]
Thank you for working on this @denkensk 

LGTM. If you can separate the generated commits from the manual changes you're making and squashing, it'll be easier to review

**Comment by alculquicondor** [comment]
/hold
for #107775

**Comment by denkensk** [comment]
> /hold
> 
> for #107775

/unhold
@Huang-Wei  please review again

**Comment by Huang-Wei** [comment]
/lgtm
/approve

Thanks!

**Comment by liggitt** [comment]
API changes lgtm. Have the stable criteria (https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/902-non-preempting-priorityclass#stable-v124) been completed?

**Comment by denkensk** [comment]
> API changes lgtm. Have the stable criteria (https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/902-non-preempting-priorityclass#stable-v124) been completed?

Yes  @liggitt 

https://github.com/kubernetes/kubernetes/pull/107775

**Comment by denkensk** [comment]
squash the commits

**Review by ravisantoshgudimetla — APPROVED** [review]
/lgtm

Leaving approval to Wei or Aldo.

**Comment by denkensk** [comment]
> There is a new commit after Wei gave his LGTM. So I thought you or Wei need to have a look at it again.

Thanks @ravisantoshgudimetla 
no updates. just squash the commits.

**Pull request merged**
